fili-core/src/main/java/com/yahoo/bard/webservice/druid/serializers/Util.java

```diff
@@ -24,24 +25,47 @@
      * @return  an Optional String of physical name
      */
     public static Optional<String> findPhysicalName(Dimension value, JsonGenerator gen) {
-        JsonStreamContext context = gen.getOutputContext();
         String apiName = value.getApiName();
         // Search for physical name
-        while (context != null) {
-            Object parent = context.getCurrentValue();
-            if (parent instanceof DruidQuery) {
-                return Optional.of(
-                        ((DruidQuery) parent)
+        return getDruidQuery(gen, druidQuery ->
+                        druidQuery
                                 .getDataSource()
                                 .getPhysicalTables()
                                 .iterator()
                                 .next()
                                 .getPhysicalColumnName(apiName)
-                );
+        );
+    }
+
+    /**
+     * JSON tree walk to determine if there is a nested query below the current json node or not.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on.
+     *
+     * @return  a Boolean where true indicates there are more nested query below this node, false otherwise
+     */
+    public static Boolean hasInnerQuery(JsonGenerator gen) {
+        return getDruidQuery(gen, druidQuery -> druidQuery.getInnerQuery() != null).get();
```

Comment:
While it's _unlikely_ that a `DruidQuery` will not be found, it's still _possible_ that this method will return an empty `Optional`. If that were to happen, `get` will throw an exception. Instead, we should handle it by either throwing our own more descriptive exception, or by using a default value instead.

In this case, I'm not sure which would be the better choice. Defaulting to `false` (ideally while logging a warning) probably wouldn't be bad, since not having an enclosing Query also means that you _can't_ have a `DataSource`, and therefore no nested query